packages/manager-react-components/src/components/datagrid/datagrid-topbar.spec.tsx

```diff
@@ -106,4 +126,106 @@ describe('datagrid topbar', () => {
     const filterListElement = screen.queryByTestId('custom-topbar-element');
     expect(filterListElement).toBeInTheDocument();
   });
+
+  it('should display visibility with all available and all shown', () => {
+    render(
+      <DatagridTopbar
+        columnsVisibility={[
+          {
+            id: 'label',
+            label: 'Label',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => true,
+            onChange: () => null,
+          },
+          {
+            id: 'price',
+            label: 'Price',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => true,
+            onChange: () => null,
+          },
+        ]}
+      />,
+    );
+
+    const visibilityElement = screen.queryByTestId(
+      'datagrid-topbar-visibility-button',
+    );
+
+    expect(visibilityElement).toHaveAttribute(
+      'label',
+      'common_topbar_columns ',
+    );
+  });
+
+  it('should display visibility with one disabled and only this disabled is shown', () => {
+    render(
+      <DatagridTopbar
+        columnsVisibility={[
+          {
+            id: 'label',
+            label: 'Label',
+            enableHiding: false,
+            isDisabled: true,
+            isVisible: () => true,
+            onChange: () => null,
+          },
+          {
+            id: 'price',
+            label: 'Price',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => false,
+            onChange: () => null,
+          },
+        ]}
+      />,
+    );
+
+    const visibilityElement = screen.queryByTestId(
+      'datagrid-topbar-visibility-button',
+    );
+
+    expect(visibilityElement).toHaveAttribute(
+      'label',
+      'common_topbar_columns (0)',
+    );
+  });
+
+  it('should display visibility with all available and one shown', () => {
+    render(
+      <DatagridTopbar
+        columnsVisibility={[
+          {
+            id: 'label',
+            label: 'Label',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => true,
+            onChange: () => null,
+          },
+          {
+            id: 'price',
+            label: 'Price',
+            enableHiding: true,
+            isDisabled: false,
+            isVisible: () => false,
+            onChange: () => null,
+          },
+        ]}
+      />,
+    );
+
+    const visibilityElement = screen.queryByTestId(
+      'datagrid-topbar-visibility-button',
+    );
+
+    expect(visibilityElement).toHaveAttribute(
+      'label',
+      'common_topbar_columns (1)',
+    );
+  });
```

Comment:
These are the test cases for Visibility Management component and Topbar component. At the Topbar components, you can only make a test to assert `VisibilityManagement` is present or not depending on the config you provide to it.